### PR TITLE
fix: takeOnFails screenshot with fullPage option fixed

### PR DESCRIPTION
**Pull request by Bayheck**
## Purpose
There is an option for testcaferc configuration file called takeOnFails. The issue here was that the screenshot taken by takeOnFails was not responding to option fullPage. The screenshot was always small. 

## Approach
I found the place where the fullPage option was not passing. Fixed it by passing fullPage option to TakeScreenshotOnFailCommand.

## References
https://github.com/DevExpress/testcafe/issues/7761

## Pre-Merge TODO
- [x] Write tests for your proposed changes
- [x] Make sure that existing tests do not fail

### CODE REVIEW

File: src/test-run/index.ts [hunk 1]
```diff
@@ -1236,7 +1236,7 @@ export default class TestRun extends AsyncEventEmitter {
         const { screenshots } = this.opts;
 
         if (!this.errScreenshotPath && (screenshots as ScreenshotOptionValue)?.takeOnFails)
-            this.errScreenshotPath = await this._internalExecuteCommand(new browserManipulationCommands.TakeScreenshotOnFailCommand({ failedActionId })) as string;
+            this.errScreenshotPath = await this._internalExecuteCommand(new browserManipulationCommands.TakeScreenshotOnFailCommand({ failedActionId, fullPage: (screenshots as ScreenshotOptionValue)?.fullPage })) as string;
```
Comment (Aleksey28) [inline src/test-run/index.ts:1239]: Please make an assignment of `screenshots` in this way `const { screenshots } = this.opts as { screenshots: ScreenshotOptionValue };`. It'll allow you to escape from the repeating `(screenshots as ScreenshotOptionValue)`.

**Review by Aleksey28 — CHANGES_REQUESTED** [review]
Please. look at my comment and check the test [Should take a screenshot on fail](https://github.com/DevExpress/testcafe/blob/21b44a0a64d30b37ea76e4837d0c989be199ef7f/test/functional/fixtures/api/es-next/take-screenshot/test.js#L680) and figure out why it didn't work.

File: src/test-run/index.ts [hunk 2]
```diff
@@ -1233,10 +1233,10 @@ export default class TestRun extends AsyncEventEmitter {
     }
 
     public async _makeScreenshotOnFail (failedActionId?: string): Promise<void> {
-        const { screenshots } = this.opts;
+        const { screenshots } = this.opts as { screenshots: ScreenshotOptionValue };
 
-        if (!this.errScreenshotPath && (screenshots as ScreenshotOptionValue)?.takeOnFails)
-            this.errScreenshotPath = await this._internalExecuteCommand(new browserManipulationCommands.TakeScreenshotOnFailCommand({ failedActionId })) as string;
+        if (!this.errScreenshotPath && screenshots?.takeOnFails)
+            this.errScreenshotPath = await this._internalExecuteCommand(new browserManipulationCommands.TakeScreenshotOnFailCommand({ failedActionId, fullPage: screenshots?.fullPage })) as string;
```
Comment (aleks-pro) [inline src/test-run/index.ts:1239]:
```suggestion
            this.errScreenshotPath = await this._internalExecuteCommand(new browserManipulationCommands.TakeScreenshotOnFailCommand({ failedActionId, fullPage: screenshots.fullPage })) as string;
```
It looks like there is no need to check if `screenshot` variable is not empty twice

File: test/server/test-run-commands-test.js [hunk 3]
```diff
@@ -3685,3 +3722,18 @@ describe('Test run commands', () => {
         });
     });
 });
+
+describe('Test run screenshot', () => {
```
Comment (aleks-pro) [inline test/server/test-run-commands-test.js:3726]: In this tests file we check command creation logic. Let's find a more suitable place for this test.

Comment (Aleksey28) [reply test/server/test-run-commands-test.js]: I think there is no need for this test after @Bayheck fixed the test 'Should take a screenshot on fail'. What do you think?

Comment (Bayheck) [reply test/server/test-run-commands-test.js]: I think so too, I've already removed this test from PR

**Review by Aleksey28 — APPROVED** [review]

**Pull request merged**